packages/core/types/@chainlink/index.d.ts

```diff
@@ -208,6 +209,7 @@ declare module '@chainlink/types' {
       | undefined
     // Optional flag to ensure adapter only uses WS and doesn't send HTTP requests
     noHttp?: boolean
+    toSaveFromFirstMessage?: (message: any) => any
```

Comment:
I'm open to a better name for this.  This function is called if we ever need to save anything from the first message received from the WS endpoint.